### PR TITLE
[master] invalidate existing tokens when deleting an oauth client

### PR DESCRIPTION
**Pull request by individual-it**
forward port of #35094 to master

### CODE REVIEW

File: apps/oauth2/tests/Controller/SettingsControllerTest.php [hunk 1]
```diff
@@ -56,18 +65,20 @@ protected function setUp(): void {
...
 		$this->settingsController = new SettingsController(
 			'oauth2',
 			$this->request,
 			$this->clientMapper,
 			$this->secureRandom,
 			$this->accessTokenMapper,
-			$l
+			$this->l,
+			$this->createMock(IAuthTokenProvider::class),
+			$this->createMock(IUserManager::class)
```
Comment (nickvergessen) [inline apps/oauth2/tests/Controller/SettingsControllerTest.php:79]: If you stored this in variables like the others above, you wouldnt need to manually create yet another controller in your test function

**Comment by nickvergessen** [comment]
The new interface should be referenced in https://github.com/nextcloud/server/issues/34692

File: apps/oauth2/tests/Controller/SettingsControllerTest.php (hunk 1 above)
Comment (individual-it) [reply apps/oauth2/tests/Controller/SettingsControllerTest.php]: @nickvergessen done in b9528076e88

**Comment by individual-it** [comment]
> The new interface should be referenced in #34692

@nickvergessen I've just updated #34692

**Comment by blizzz** [comment]
conflicts

**Comment by individual-it** [comment]
@blizzz conflicts solved

**Review by come-nc — APPROVED** [review]
I rebased on master and cleaned up the composer diff to make the diff readable.

@nickvergessen It’s not clear what to do with the OCP part, because it was merged in stable24 and not the other branches, what do we put in the `@since`? Should I leave 24.0.8 or should I put 26.0.1 or something?

**Comment by nickvergessen** [comment]
> Should I leave 24.0.8 or should I put 26.0.1 or something?

Multiple since are supported, so we could have 24.0.x (.8 if it gets merged and backported in time), 25.0.y and 26.0.z or just either of them. Since apps can not require maintenance versions at the moment, they have to do defensive programming anyway, so not too much of an issue.

**Review by julien-nc — APPROVED** [review]

**Pull request merged**